src/rm_cms.c

```diff
@@ -0,0 +1,309 @@
+#include <strings.h>        // strncasecmp
+#include <stdlib.h>         // malloc
+#include <math.h>           // ceil, log10f
+
+#include "version.h"
+#include "rmutil/util.h"
+
+#include "rm_cms.h"
+#include "cms.h"
+
+RedisModuleType *CMSketchType;
+
+typedef struct {
+    const char *key;
+    long long value;
+} CMSPair;
+
+static int CompareCStringToRedisString(char *str, RedisModuleString *redisStr) {
+    size_t len;
+    char *cRedisStr = (char *)RedisModule_StringPtrLen(redisStr, &len);
+    return strncasecmp(cRedisStr, str, len) == 0;
+}
+
+static int GetCMS(RedisModuleCtx *ctx, RedisModuleString *keyName, 
+                    CMSketch **cms, int mode) {
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, mode);
+    if(RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
+        RedisModule_ReplyWithError(ctx, "CMS: key does not exist");
+        return REDISMODULE_ERR;
```

Comment:
This should be REDISMODULE_OK. _ERR is reserved for real errors and, afair, isn't really used atm. Same goes for the next return.

Reply:
This is for internal use. The function which calls it returns _OK

Reply:
If for internal use just return 0 or 1, or make your own defines... imo. The current is kind of confusing (to me at least).

Reply:
👍

Reply:
In RediSearch and elsewhere, we use REDISEARCH_OK/REDISEARCH_ERR as internal return codes all the time. Nothing wrong with that as long as it's consistent and documented. I believe they expand to the same 0 and 1 anyway, though.

Reply:
Added INNER_ERROR with both RedisModule_ReplyWithError and ERROR